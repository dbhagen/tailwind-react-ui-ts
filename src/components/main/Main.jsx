import React from 'react'
import PropTypes from 'prop-types'

import { Box } from '../primitives'

const Main = ({ children, is, id, ...rest }) => (
  <Box is={is} id={id} role="main" {...rest}>
    {children}
  </Box>
)

Main.propTypes = {
  children: PropTypes.node,
  id: PropTypes.string,
  is: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]),
}

Main.defaultProps = {
  children: undefined,
  id: 'main',
  is: 'main',
}

export default Main
