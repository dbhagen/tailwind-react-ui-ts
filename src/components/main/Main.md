Simple wrapper around the standard `<main />` element which adds an ID for use with an access friendly skip link.

### Example

Tab onto the example below to see this behaviour in action.

```jsx
<SkipLink href="#main-example" />
<Card h={64} bg="gray-500" text="white" flex items="center" justify="center" rounded="none">
  Some interceding content, e.g. a header / navigation
</Card>
<Main id="main-example" p={4} border rounded="b">
  <Text>This is the main body of my app... disappointing isn't it.</Text>
</Main>
```

Note that the ID's were set for this example to avoid conflict with the auto-generated styleguide. With no values specificied the components will default to using `main`.
