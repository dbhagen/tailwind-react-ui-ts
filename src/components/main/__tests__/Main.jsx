import React from 'react'
import { shallow } from 'enzyme'

import Main from '../Main'

const setup = (testProps = {}) => {
  const props = Object.assign(testProps)

  const wrapper = shallow(<Main {...props}>Lorem ipsum</Main>)

  return {
    props,
    wrapper,
  }
}

describe('Main', () => {
  it('renders matching snapshot', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
})
