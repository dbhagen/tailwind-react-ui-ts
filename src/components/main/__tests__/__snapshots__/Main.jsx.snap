// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders matching snapshot 1`] = `
<Box
  id="main"
  inline={false}
  inlineBlock={false}
  is="main"
  role="main"
>
  Lorem ipsum
</Box>
`;
